### DISCUSSION TITLE
ignore an commit, when it is has been remdiated?

### DISCUSSION THREAD

**Discussion by dileepdkumar** [zricethezav/gitleaks#1092] [Q&A]
How to ignore an commit (which has an commit with private key), when it is has been remediated in later commits in the git leaks secret scanner?

**Comment by maltemorgenstern** [comment]
Hey @dileepdkumar,
I am not sure if I understand your question correctly - if not, then please clarify.

I understand that you have a repository that contains a commit which leaked a secret (e.g. added a private key). After you realized that you immediately pushed a new commit that removed the private key from the repo.

Running `gitleaks` now is still going to report the "old" commit - because it leaked the private key. If you want to resolve/suppress this finding then take a look at the `.gitleaksignore` file: https://github.com/gitleaks/gitleaks#gitleaksignore

It allows you to suppress specific findings based on the commit hash and secret location. In your case this could look like this:
```
2186f2299990290e4e19cf3b28906e2ec369d01e:my-secret-key.ed25519:private-key:1
```
Another example can be found in this repository - here is the [`.gitleaksignore` entry](https://github.com/gitleaks/gitleaks/blob/765bcd5854535a4e4d40eb423eea3c1bec9c5746/.gitleaksignore#L656) and here the ["leaked" private key](https://github.com/gitleaks/gitleaks/blob/6d801ed61cf4d75d176cbf9062ec6ad706c9d1eb/testdata/repos/symlinks/source_file/id_ed25519).

I hope this helps you resolving/suppressing any findings!

Regards
Malte
